src/Factions.cpp

```diff
@@ -544,8 +566,12 @@ const Faction* FactionsDatabase::GetNearestFaction(const Sector::System* sys) co
 	double closestFactionDist = HUGE_VAL;
 	ConstFactionList& candidates = m_spatial_index.CandidateFactions(sys);
 
-	for (ConstFactionIterator it = candidates.begin(); it != candidates.end(); ++it) {
-		if ((*it)->IsCloserAndContains(closestFactionDist, sys)) result = *it;
+	for (ConstFactionIterator it = candidates.begin(); it != candidates.end(); ++it)
+	{
+		if ((*it)->IsClaimed(sys->GetPath()))
```

Comment:
Granted, I haven't digged into the code here, so my understanding is very low (sorry) , but just at a glance I assumed this would be an `if-else`, not `if-if`, so the isClaimed should take precedence over what ever was there before?

Reply:
yeah youre right

Reply:
In the fix, to replace it with a `return`, you're breaking out of the for-loop, which, at a glance, looks unsymmetric with the other if-statment in the loop, so just checking if it is right that is should break out of the for-loop if `isClaimed`?

Reply:
Yes, it's correct: as I understand it, any explicit claim takes over any proximity-induced implicit claim.

However, why is this mechanism in a function called "GetNearestFaction" ? The name implies very specific semantics, straying from those isn't a very good idea and will lead to confusion further down the road.

Reply:
GetGeneratedFaction, FindBestFaction, ... hmm err whats suitable?

Reply:
function is called twice, once by AssignFaction, once by StarSystemFromSectorGenerator::Apply,
so.. call it FindFactionToApply? GetFactionToApply?